File: Practise/stack_using_array.py
class Stack:
    def __init__(self,size=100):
        self.stack = []
        self.size = size

    def push(self,val):
        if len(self.stack) == self.size:
            print("Stack Overflow")
        else:
            self.stack.append(val)
            print(f"{val} pushed")

    def pop(self):
        if not self.stack:
            print("Stack Underflow")
        else:
            val = self.stack.pop()
            print(f"{val} popped")

    def display(self):
        if not self.stack:
            print("Stack is empty")
        else:
            print("Stack elements: ",self.stack[::-1])

    def peek(self):
        if not self.stack:
            print("Stack is empty")
            return None
        return self.stack[-1]

    def isEmpty(self):
        return len(self.stack) == 0


s = Stack(5)
s.push(10)
s.push(20)
s.push(30)
s.display()
s.pop()
s.display()

File: Practise/stack_using_array.c
#include <stdio.h>
#define MAX 100

int stack[MAX];
int top = -1;

void push(int val){
    if(top == MAX-1){
        printf("Stack Overflow\n");
    }
    else{
        stack[++top] = val;
        printf("%d pushed\n",val);
    }
}

void pop(){
    if(top == -1){
        printf("Stack Underflow");
    }
    else{
        printf("%d popped\n",stack[top]);
        stack[top--];
    }
}

void display(){
    if(top == -1){
        printf("Stack is empty\n");
    }
    else{
        printf("Stack elements: ");
        for(int i=top;i>=0;i--){
            printf("%d ",stack[i]);
        }
        printf("\n");
    }
}

int peek(){
    if(top == -1){
        printf("Stack is empty");
        return -1;
    }
    else{
        return stack[top];
    }
}

int isEmpty(){
    return (top == -1);
}

int main() {
    push(10);
    push(20);
    push(30);
    display();
    pop();
    display();
    return 0;
}